src/ApplicationFactory.php

```diff
@@ -30,33 +26,26 @@
  */
 final class ApplicationFactory
 {
-    public function __invoke(ContainerInterface $container): Application
-    {
-        $config = $container->get('config')['laminas-cli'] ?? [];
-        Assert::isMap($config);
+    public const CONTAINER_OPTION = 'container';
 
+    public function __invoke(): Application
+    {
         /** @psalm-suppress DeprecatedClass */
         $version = strstr(Versions::getVersion('laminas/laminas-cli'), '@', true);
         Assert::string($version);
-
-        $commands = $config['commands'] ?? [];
-        Assert::isMap($commands);
-        Assert::allString($commands);
-
-        $eventDispatcherServiceName = __NAMESPACE__ . '\SymfonyEventDispatcher';
-        $dispatcher                 = $container->has($eventDispatcherServiceName)
-            ? $container->get($eventDispatcherServiceName)
-            : new EventDispatcher();
-        Assert::isInstanceOf($dispatcher, EventDispatcherInterface::class);
-
-        $dispatcher->addListener(ConsoleEvents::TERMINATE, new TerminateListener($config));
-
         $application = new Application('laminas', $version);
-        // phpcs:ignore WebimpressCodingStandard.PHP.CorrectClassNameCase
-        $application->setCommandLoader(new ContainerCommandLoader($container, $commands));
-        $application->setDispatcher($dispatcher);
         $application->setAutoExit(false);
 
+        $definition = $application->getDefinition();
+        $definition->addOption(
+            new InputOption(
+                self::CONTAINER_OPTION,
+                'c',
```

Comment:
Let's not define a short-name for this. "c" is something I can see user commands wanting to use for an option (I can think of at least two commands I've written personally that use this short name off the top of my head).